File: Source/Rio.Models/DataTransferObjects/Offer/OfferDto.cs
using Rio.Models.DataTransferObjects.Account;
using System;

namespace Rio.Models.DataTransferObjects.Offer
{
    public class OfferDto
    {
        public int OfferID { get; set; }
        public DateTime OfferDate { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public string OfferNotes { get; set; }
        public AccountDto CreateAccount { get; set; }
        public OfferStatusDto OfferStatus { get; set; }
        public int TradeID { get; set; }
        public int? WaterTransferID { get; set; }
    }
}

File: Source/Rio.Web/src/app/shared/models/offer/offer-dto.ts
import { OfferStatusDto } from './offer-status-dto';
import { AccountSimpleDto } from '../account/account-simple-dto';
import { TradeDto } from './trade-dto';

export class OfferDto {
    OfferID: number;
    OfferDate: Date;
    Quantity: number;
    Price: number;
    OfferNotes: string;
    CreateAccount: AccountSimpleDto;
    OfferStatus: OfferStatusDto;
    Trade: TradeDto;
    WaterTransferID: number;

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

